snapshot.py

```diff
@@ -0,0 +1,44 @@
+import json
+import subprocess
+
+def establish_ssh_connectivity():
+    node_ip_address_list = []
+    command = ["oc", "get", "nodes", "-o", "json"]
+    node_output=subprocess.check_output(command)
+
+    node_map=json.loads(node_output)
+    for item in node_map["items"]:
+        if "node-role.kubernetes.io/control-plane" in item["metadata"]["labels"]:
+            node_ip_address_list.append(item["status"]["addresses"][0]["address"])
+
+    for node_ip_address in node_ip_address_list:
+        command = ["ssh", "-o", "StrictHostKeyChecking=no", f"core@{node_ip_address}", "true"]
+        print(subprocess.run(command))
+
+
+def get_cp_names():
+    node_name_list = []
+    command = ["oc", "get", "nodes", "-o", "json"]
+    node_output=subprocess.check_output(command)
+
+    node_map=json.loads(node_output)
+    for item in node_map["items"]:
+        if "node-role.kubernetes.io/control-plane" in item["metadata"]["labels"]:
+            node_name_list.append(item["metadata"]["name"])
+    return node_name_list
+
+def backup_cluster(true_master_node):
+    try:
+        command = ["oc", "debug", f"node/{true_master_node}", "--", "chroot", "/host", "/bin/bash", "/usr/local/bin/cluster-backup.sh","/home/core/assets/backup"]
+        print(subprocess.run(command))
+    except:
+        print("backup failed")
+
+establish_ssh_connectivity()
```

Comment:
It feels like establish_ssh_connectivity() and get_cp_names() could be simplified. Maybe get the cp names, then pass the names to a establish_ssh_connectivity to verify this is working.

Also, what is "cp" names exactly?

Reply:
ack.

it is contol plane